paper-server/src/main/java/io/papermc/paper/command/PaperVersionCommand.java

```diff
@@ -0,0 +1,183 @@
+import net.minecraft.server.MinecraftServer;
+import org.bukkit.Bukkit;
+import org.bukkit.command.CommandSender;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.util.StringUtil;
+import org.jspecify.annotations.NullMarked;
+
+@NullMarked
+public class PaperVersionCommand {
+    public static final String DESCRIPTION = "Gets the version of this server including any plugins in use";
+
+    private static final Component NOT_RUNNING = Component.text()
+        .append(Component.text("This server is not running any plugin by that name."))
+        .appendNewline()
+        .append(Component.text("Use /plugins to get a list of plugins.").clickEvent(ClickEvent.suggestCommand("/plugins")))
+        .build();
+    private static final JoinConfiguration PLAYER_JOIN_CONFIGURATION = JoinConfiguration.separators(
+        Component.text(", ", NamedTextColor.WHITE),
+        Component.text(", and ", NamedTextColor.WHITE)
+    );
+
+    private final VersionFetcher versionFetcher = new PaperVersionFetcher();
+    private CompletableFuture<ComputedVersion> computedVersion = CompletableFuture.completedFuture(new ComputedVersion(Component.empty(), -1)); // Precompute-- someday move that stuff out of bukkit
+
+    public static LiteralCommandNode<CommandSourceStack> create() {
+        final PaperVersionCommand command = new PaperVersionCommand();
+
+        return Commands.literal("version")
+            .requires(source -> source.getSender().hasPermission("bukkit.command.version"))
+            .then(Commands.argument("plugin", StringArgumentType.word())
+                .suggests(command::suggestPlugins)
+                .executes(command::pluginVersion))
+            .executes(command::serverVersion)
+            .build();
+    }
+
+    private int pluginVersion(final CommandContext<CommandSourceStack> context) {
+        final CommandSender sender = context.getSource().getSender();
+        final String pluginName = context.getArgument("plugin", String.class);
+
+        Plugin plugin = Bukkit.getPluginManager().getPlugin(pluginName);
+        if (plugin == null) {
+            plugin = Arrays.stream(Bukkit.getPluginManager().getPlugins())
+                .filter(checkPlugin -> checkPlugin.getName().toLowerCase(Locale.ROOT).contains(pluginName))
+                .findAny()
+                .orElse(null);
+        }
+
+        if (plugin != null) {
+            this.sendPluginInfo(plugin, sender);
+        } else {
+            sender.sendMessage(NOT_RUNNING);
+        }
+
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private CompletableFuture<Suggestions> suggestPlugins(final CommandContext<CommandSourceStack> context, final SuggestionsBuilder builder) {
+        for (final Plugin plugin : Bukkit.getPluginManager().getPlugins()) {
+            final String name = plugin.getName();
+            if (StringUtil.startsWithIgnoreCase(name, builder.getRemainingLowerCase())) {
```

Comment:
No need to getRemainingLowerCase, else put it into an extra variable outside the loop